### PR TITLE
Update for network related tests

### PR DESCRIPTION
**Pull request by kylazhang**
The new update for net-update cases related to this patch
1. add index for several functions as in network xml it can have several ip sections
2. add connection for network xml
3. change the get_active, del_active func for undefined check as undefined network
also can be actived and be deactived
4. add function del_element and get_interface_connection
5. change get_net_if for getting more judgement to get interface
6. add timeout for get_guest_ip_addr

### CODE REVIEW

File: virttest/libvirt_xml/base.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def copy(self):
             pass  # no XML was loaded yet
         return the_copy
 
-    def get_section_string(self, xpath):
+    def get_section_string(self, xpath, index=0):
```
Comment (chunfuwen) [inline virttest/libvirt_xml/base.py]: add doc string here

File: virttest/libvirt_xml/network_xml.py [hunk 2]
```diff
@@ -484,10 +486,12 @@ def del_defined(self):
 
     def get_active(self):
         """Accessor method for 'active' property (True/False)"""
-        self.__check_undefined__("Cannot determine activation for undefined "
-                                 "network")
         state_dict = self.virsh.net_state_dict(virsh_instance=self.virsh)
-        return state_dict[self.name]['active']
+        try:
+            active_state = state_dict[self.name]['active']
```
Comment (chunfuwen) [inline virttest/libvirt_xml/network_xml.py]: return state_dict[self.name]['active'] directly, and line 494 can be leave out

Comment (kylazhang) [reply virttest/libvirt_xml/network_xml.py]: It is for catching error

File: virttest/libvirt_xml/base.py (hunk 1 above)
Comment (kylazhang) [reply virttest/libvirt_xml/base.py]: ok

File: virttest/utils_net.py [hunk 3]
```diff
@@ -1288,22 +1288,18 @@ def local_runner_status(cmd, timeout=None, shell=False):
     return process.run(cmd, verbose=False, timeout=timeout, shell=shell).exit_status
 
 
-def get_net_if(runner=None, state=None):
+def get_net_if(runner=local_runner, state=".*", qdisc=".*", optional=".*"):
     """
     :param runner: command runner.
     :param div_phy_virt: if set true, will return a tuple division real
```
Comment (hahakiki2010) [inline virttest/utils_net.py]: The doc string seemed outdate

File: virttest/utils_net.py [hunk 4]
```diff
@@ -1418,41 +1414,47 @@ def get_guest_ip_addr(session, mac_addr, os_type="linux", ip_version="ipv4",
...
+                nic_address = get_net_if_addrs_win(session, mac_addr)
             else:
-                return [x for x in nic_address[ip_version]
-                        if not x.lower().startswith(linklocal_prefix)][0]
-        except IndexError:
-            return None
-    except Exception as err:
-        logging.debug(session.cmd_output(info_cmd))
-        raise IPAddrGetError(mac_addr, err)
+                info_cmd = ""
```
Comment (hahakiki2010) [inline virttest/utils_net.py]: Line 1434 seemed duplicate with Line 1420

File: virttest/libvirt_xml/network_xml.py [hunk 5]
```diff
@@ -580,6 +583,23 @@ def del_ip(self):
...
+        """
+        Delete element from network xml
+
+        :param element: xpath of element like '/ip/dhcp'
+        :param index: index of element that want to delete
+        """
+        xmltreefile = self.__dict_get__('xml')
+        try:
+            del_elem = xmltreefile.findall(element)[index]
+        except IndexError as detail:
```
[the remark this answers is not in the document]
Comment (kylazhang) [reply virttest/libvirt_xml/network_xml.py]: @hahakiki2010  findall will return [] if not exist, then it will raise IndexError

**Pull request merged**